pkg/commands/kustomizationfile.go

```diff
@@ -30,9 +34,18 @@ import (
 	"github.com/kubernetes-sigs/kustomize/pkg/types"
 )
 
+var kustomizationFields = []string{"resources", "bases", "namePrefix", "namespace", "crds", "commonLabels", "commonAnnotations", "patches", "configMapGenerator", "secretGenerator", "vars", "imageTags"}
+
+type fieldComment struct {
```

Comment:
please called this `commentedField`, and add a comment to the type saying something like `field` has to be a recognized kustomization field, but `comment` can be empty.

Reply:
added.